Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -191,151 +275,22 @@ CFArrayRef _DWriteCopyFontNamesForFamilyName(CFStringRef familyName) {
-                                                    { CFSTR("NORMAL"), DWRITE_FONT_WEIGHT_NORMAL },
-                                                    { CFSTR("REGULAR"), DWRITE_FONT_WEIGHT_REGULAR },
-                                                    { CFSTR("MEDIUM"), DWRITE_FONT_WEIGHT_REGULAR },
-                                                    { CFSTR("DEMIBOLD"), DWRITE_FONT_WEIGHT_DEMI_BOLD },
-                                                    { CFSTR("DEMI"), DWRITE_FONT_WEIGHT_DEMI_BOLD },
-                                                    { CFSTR("SEMIBOLD"), DWRITE_FONT_WEIGHT_SEMI_BOLD },
-                                                    { CFSTR("BOLD"), DWRITE_FONT_WEIGHT_BOLD },
-                                                    { CFSTR("EXTRABOLD"), DWRITE_FONT_WEIGHT_EXTRA_BOLD },
-                                                    { CFSTR("ULTRABOLD"), DWRITE_FONT_WEIGHT_ULTRA_BOLD },
-                                                    { CFSTR("BLACK"), DWRITE_FONT_WEIGHT_BLACK },
-                                                    { CFSTR("HEAVY"), DWRITE_FONT_WEIGHT_HEAVY },
-                                                    { CFSTR("EXTRABLACK"), DWRITE_FONT_WEIGHT_EXTRA_BLACK },
-                                                    { CFSTR("ULTRABLACK"), DWRITE_FONT_WEIGHT_ULTRA_BLACK } };
-
 /**
  * Helper that parses a font name, and returns appropriate weight, stretch, style, and family name values
  */
-_DWriteFontProperties _DWriteGetFontPropertiesFromName(CFStringRef fontName) {
-    // Set some defaults for when weight/stretch/style are not mentioned in the name
-    _DWriteFontProperties ret{ DWRITE_FONT_WEIGHT_NORMAL,
-                               DWRITE_FONT_STRETCH_NORMAL,
-                               DWRITE_FONT_STYLE_NORMAL,
-                               _DWriteGetFamilyNameForFontName(fontName) };
-
-    // Relationship of family name -> font name not always consistent
-    // Usually, properties are added to the end (eg: Arial -> Arial Narrow Bold)
-    // However, this is not always the case (eg: Eras ITC -> Eras Bold ITC)
-    // In addition, some fonts with properties are occasionally placed into their own family (eg: Segoe WP SemiLight -> Segoe WP SemiLight)
-    // Try to be more prudent about these edge cases, by looking only at the difference between the font name and family name
-    woc::unique_cf<CFArrayRef> fontNameTokens(CFStringCreateArrayBySeparatingStrings(kCFAllocatorDefault, fontName, CFSTR(" ")));
-    woc::unique_cf<CFMutableSetRef> propertyTokens(CFSetCreateMutable(kCFAllocatorDefault, 0, &kCFTypeSetCallBacks));
-
-    for (size_t i = 0; i < CFArrayGetCount(fontNameTokens.get()); ++i) {
-        CFSetAddValue(propertyTokens.get(), CFArrayGetValueAtIndex(fontNameTokens.get(), i));
-    }
+std::shared_ptr<_DWriteFontProperties> _DWriteGetFontPropertiesFromName(CFStringRef fontName) {
+    woc::unique_cf<CFLocaleRef> locale(CFLocaleCopyCurrent());
```

Comment:
locale can change through the life of an app.

---
In reply to: [87664268](https://github.com/Microsoft/WinObjC/pull/1366#discussion_r87664268) [](ancestors = 87664268)